source/opt/folding_rules.cpp

```diff
@@ -2539,6 +2532,36 @@ FoldingRule RedundantIAdd() {
   };
 }
 
+// This rule handles any of RedundantBinaryRhs0Ops with a 0 or vector 0 on the
+// right-hand side.
+static const constexpr spv::Op RedundantBinaryRhs0Ops[] = {
+    spv::Op::OpBitwiseOr,
+    spv::Op::OpBitwiseXor,
+    spv::Op::OpShiftRightLogical,
+    spv::Op::OpShiftRightArithmetic,
+    spv::Op::OpShiftLeftLogical,
+    spv::Op::OpIAdd,
+    spv::Op::OpISub};
+FoldingRule RedundantBinaryRhs0(spv::Op op) {
+  assert(std::find(std::begin(RedundantBinaryRhs0Ops),
+                   std::end(RedundantBinaryRhs0Ops),
+                   op) != std::end(RedundantBinaryRhs0Ops) &&
+         "Wrong opcode.");
+  return RedundantBinaryOpWithZeroOperand(1);
+}
+
+// This rule handles any of RedundantBinaryLhs0Ops with a 0 or vector 0 on the
+// left-hand side.
+static const constexpr spv::Op RedundantBinaryLhs0Ops[] = {
+    spv::Op::OpBitwiseOr, spv::Op::OpBitwiseXor, spv::Op::OpIAdd};
+FoldingRule RedundantBinaryLhs0(spv::Op op) {
```

Comment:
I'm seeing this error in the release builds:

```


```
/tmpfs/src/github/SPIRV-Tools/source/opt/folding_rules.cpp:2545:41: error: unused parameter 'op' [-Werror,-Wunused-parameter]
FoldingRule RedundantBinaryRhs0(spv::Op op) {
                                        ^
/tmpfs/src/github/SPIRV-Tools/source/opt/folding_rules.cpp:2557:41: error: unused parameter 'op' [-Werror,-Wunused-parameter]
FoldingRule RedundantBinaryLhs0(spv::Op op) {
```
```

```suggestion
FoldingRule RedundantBinaryLhs0(spv::Op op) {
  (void)(op);
```